core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -448,6 +453,29 @@ class TimeSeriesShard(val dataset: Dataset,
 
   def indexValues(indexName: String, topK: Int): Seq[TermInfo] = partKeyIndex.indexValues(indexName, topK)
 
+  def metadata(filter: Seq[ColumnFilter], columns: Seq[String], startTime: Long, endTime: Long):
+          Set[Map[ZeroCopyUTF8String, ZeroCopyUTF8String]] = {
+    val result = new HashSet[Map[ZeroCopyUTF8String, ZeroCopyUTF8String]]
+    if (filter.isEmpty) {
+      columns.foreach(column => {
+        result += partKeyIndex.indexValues(column, 1000)
```

Comment:
Wait, so this API returns terms from multiple tags and partition keys from filters?  You really can't use the same method to do both.  You need different methods.

Actually we just really need one method: to return partition keys given filters.   The other method in this part of the code is IMO not necessary.    There is already an `indexValues` method.  That method can just be composed multiple times to get lists for different columns/tags.  That composition should be done in the query LeafExecPlan, not in the TimeSeriesShard.

Reply:
Actually, my new thought is that this entire method is not needed at all.

- For finding top K values given column names, just use the existing `indexValues` method and compose results for different columns in the ExecPlan directly.  
- For finding partition keys given filters, just use the `scanPartitions` method, and extract the partition keys from each `ReadablePartition` that is returned.   We might need a new `ChunkScanMethod` which is `NoChunks`.